src/MoBi.Core/Commands/UpdateSolverAndSchemaInSimulationCommand.cs

```diff
@@ -0,0 +1,53 @@
+using MoBi.Assets;
+using MoBi.Core.Domain.Model;
+using MoBi.Core.Events;
+using OSPSuite.Assets;
+using OSPSuite.Core.Commands.Core;
+using OSPSuite.Core.Domain;
+
+namespace MoBi.Core.Commands
+{
+   public class UpdateSolverAndSchemaInSimulationCommand : SimulationChangeCommandBase
+   {
+      private SolverSettings _newSolverSettings;
+      private OutputSchema _newOutputSchema;
+      private byte[] _serializedOldSolverSettings;
+      private byte[] _serializedOldOutputSchema;
+
+      public UpdateSolverAndSchemaInSimulationCommand(IMoBiSimulation simulation, SolverSettings solverSettings, OutputSchema outputSchema) : base(simulation)
+      {
+         CommandType = AppConstants.Commands.UpdateCommand;
+         ObjectType = ObjectTypes.SimulationSettings;
+         Description = AppConstants.Commands.UpdateSimulationSolverSettingsAndSchemaInSimulation(simulation.Name);
+         _newSolverSettings = solverSettings;
+         _newOutputSchema = outputSchema;
+      }
+
+      protected override ICommand<IMoBiContext> GetInverseCommand(IMoBiContext context)
+      {
+         return new UpdateSolverAndSchemaInSimulationCommand(_simulation, context.Deserialize<SolverSettings>(_serializedOldSolverSettings), context.Deserialize<OutputSchema>(_serializedOldOutputSchema));
```

Comment:
I don't quite get the whole serialization deserialization stuff. Whu aren't ou just using a reference to the old objects?

Reply:
there is a method that is called before the inverse is created to resolve object. This is where the oldSsettings and schema should be resolves instead of in the GeInverseCommand IMO